corehq/apps/style/static/style/js/hq_extensions.jquery.js

```diff
@@ -44,9 +44,43 @@
     $.fn.closest_form = function () {
         return this.closest('form, .form');
     };
+
     $.fn.my_serialize = function () {
         var data = this.find('[name]').serialize();
         return data;
     };
 
+    $.fn.addSpinnerToButton = function () {
+        $(this).prepend('<i class="fa fa-refresh fa-spin icon-refresh icon-spin"></i> ');
+    };
+
+    $.fn.removeSpinnerFromButton = function () {
+        $(this).find('i').remove();
+    };
+
+    $.fn.disableButtonNoSpinner = function () {
+        $(this).attr('disabled', 'disabled')
+            .addClass('disabled');
+    };
+
+    $.fn.disableButton = function () {
+        $(this).disableButtonNoSpinner();
+        $(this).addSpinnerToButton();
+    };
+
+    $.fn.enableButton = function () {
+        $(this).removeSpinnerFromButton();
+        $(this).removeClass('disabled')
+            .removeAttr('disabled');
+    };
+
+    $.fn.koApplyBindings = function (context) {
```

Comment:
You definitely can do that if you break these files apart.  Here's what that might look like:
```javascript
// file 1
define(['jquery', 'jquery.form', 'bootstrap'], function($, ignore1, ignore2) {});
// file 2
define(['jquery', 'bootstrap', 'jquery.cookie', 'jquery.hq'], function($, ignore1, ignore2, ignore3) {});
```
The chief advantage here is that you can be as granular as you like with resources.

The way I've implemented it is as a monolith, so in both cases it'd just be 
```javascript
define(['hquery'], function($) {});
```
The advantages of this are:
* simpler to write and remember
* easier to adapt existing code (it's tricky to tell which jquery utils are provided by which plugins)
* `$` means the same thing in all contexts - if it works in one place, it'll work in another.

Also, you can always fall back to the former way if you have a good reason to avoid using the monolithic `hquery`